packages/api-reference/src/hooks/useNavigate.ts

```diff
@@ -0,0 +1,88 @@
+import { slug } from 'github-slugger'
+import { ref } from 'vue'
+
+import { type Heading, sleep } from '../helpers'
+import type { Tag, TransformedOperation } from '../types'
+
+type NavState = {
+  id?: string
```

Comment:
Does this encapsulate the whole state? Can/should we break this down further into the individual elements of tag/path/method/operation?